internal/store/memory_restart_rule_store.go

```diff
@@ -135,3 +150,96 @@
 
 	return false
 }
+
+// delayKey creates a consistent key for delayed restart tracking
+func (s *MemoryRestartRuleStore) delayKey(workloadKey, workloadKind string) string {
+	return fmt.Sprintf("%s/%s", workloadKind, workloadKey)
+}
+
+// IsWorkloadDelayed checks if a workload is currently scheduled for delayed restart
+func (s *MemoryRestartRuleStore) IsWorkloadDelayed(ctx context.Context, workloadKey, workloadKind string) bool {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	key := s.delayKey(workloadKey, workloadKind)
+	delayedRestart, exists := s.delayedRestarts[key]
+	if !exists {
+		return false
+	}
+
+	// Check if the delay has expired
+	if time.Now().After(delayedRestart.RestartAt) {
+		// Delay has expired, remove it
+		delete(s.delayedRestarts, key)
+
+		return false
+	}
+
+	return true
+}
+
+// AddDelayedRestart schedules a workload for delayed restart
+// If the workload is already delayed, it extends the delay if the new delay is longer
+func (s *MemoryRestartRuleStore) AddDelayedRestart(ctx context.Context, workloadKey, workloadKind string, delay time.Duration) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	key := s.delayKey(workloadKey, workloadKind)
+	now := time.Now()
+	newRestartAt := now.Add(delay)
+
+	// Check if there's already a delayed restart for this workload
+	if existingDelay, exists := s.delayedRestarts[key]; exists {
+		// Use the longer delay (later restart time)
```

Comment:
This is not good. It could delay the restart indefinitely if changes occur frequent and delay is big